src/libraries/Microsoft.Extensions.Primitives/src/ChangeToken.cs

```diff
@@ -142,7 +142,7 @@ public void Dispose()
             {
                 // If the previous value is disposable then dispose it, otherwise,
                 // now we've set the disposed sentinel
-                Interlocked.Exchange(ref _disposable, _disposedSentinel).Dispose();
+                Interlocked.Exchange(ref _disposable, _disposedSentinel)?.Dispose();
```

Comment:
I don't believe `_disposable` can ever be `null`, and thus this should never need to check for `null`.

The ctor of this class calls `RegisterChangeTokenCallback`, which calls `SetDisposable`, which always ensures `_disposable` is non-null.

Reply:
My mistake - looking more at this, I see `RegisterChangeTokenCallback` checks `if (token is null)`, but that doesn't match the nullabillity of the `IChangeToken token` parameter. Does the signature of `RegisterChangeTokenCallback` need to be updated to allow for a `null` token?

Reply:
Does it make sense to call `ChangeToken.OnChange` and not subscribe to a change? Considering `RegisterChangeTokenCallback` doesn't throw an exception, maybe it is.

Reply:
I'd prefer we did product changes separately mainly for two reasons:
- less chance to add any regression - in here you'll not regress since you're just adding a null check but it opens a window of temptation for more and more complex tiny refactoring which could regress and it's easier to just not allow any product changes in nullability PR and focus on checking annotations rather than "is the product change correct" on top of nullable annotations
- null check PR will be a separate commit and will be easier to find if we search for "what changes have been made to Microsoft.Extensions.Primitives" - this is better in case we need to backport some behavior or tracking some